command/run.go

```diff
@@ -208,22 +251,49 @@ func (c *RunCommand) Run(args []string) int {
 		return Success
 	}
 
-	resultsFile := a.ResultsDest()
-	if err = writeSummary(os.Stdout, resultsFile, a.ManifestOps); err != nil {
+	if err = writeSummary(teeWriter, tmp, a.ManifestOps); err != nil {
 		l.Warn("failed to generate report summary; please review output files to ensure everything expected is present", "err", err)
-		return RunError
+		return OutputError
+	}
+
+	// Include a timestamp based on agent start time
+	// specifically excluding colons ":" since they are anathema to some filesystems and programs.
+	ts := a.Start.UTC().Format("2006-01-02T150405Z")
+	err = compressOutputDir(tmp, "hcdiag"+ts, a.Config.Destination)
+	if err != nil {
+		l.Warn("failed to compress output directory; please review output files", "tmpdir", tmp, "err", err)
+		return OutputError
 	}
 
 	return Success
 }
 
+func compressOutputDir(tmpDir, filename, destination string) error {
+	err := util.EnsureDirectory(destination)
+	if err != nil {
+		return fmt.Errorf("Failed to ensure destination directory exists: dir=%s, error=%w", destination, err)
+	}
+
+	// Build bundle destination path from config
+	resultsFile := fmt.Sprintf("%s.tar.gz", filename)
+	resultsDest := filepath.Join(destination, resultsFile)
+
+	// Archive and compress outputs
+	err = util.TarGz(tmpDir, resultsDest, filename)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 // configureLogging takes a logger name, sets the default configuration, grabs the LOG_LEVEL from our ENV vars, and
 // returns a configured and usable logger.
-func configureLogging(loggerName string) hclog.Logger {
+func configureLogging(loggerName string, loggerOutput io.Writer) hclog.Logger {
 	// Create logger, set default and log level
 	appLogger := hclog.New(&hclog.LoggerOptions{
-		Name:  loggerName,
-		Color: hclog.AutoColor,
+		Name: loggerName,
+		// Color:  hclog.AutoColor,
```

Comment:
Looks like some commented code

Reply:
Yeah, this change currently breaks coloring but today I'll look at wrapping this logger in our logging functions to fix that.

Reply:
Ah! That makes sense.